### PR TITLE
add image formats and validation for ImageContent 

### PR DESCRIPTION
**Pull request by ndoriomedov**

### CODE REVIEW

File: src/Kernel/Validators/ImageContentValidator.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        {
+            RuleFor(content => content)
+                .NotEmpty().WithMessage("Content can't be empty.");
+
+            When(content => !string.IsNullOrEmpty(content),
+            () =>
+            {
+                RuleFor(content => content)
+                    .Must(content => Convert.TryFromBase64String(content, new Span<byte>(new byte[content.Length]), out _))
+                    .WithMessage("Content must be base64 string");
```
Comment (VLiamin) [inline src/Kernel/Validators/ImageContentValidator.cs]: May be add dot after message.

**Review by VLiamin — APPROVED** [review]

File: src/Kernel/Database/IBaseDataProvider.cs [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
     /// Base data provider interface.
     /// </summary>
     public interface IBaseDataProvider
     {
         /// <summary>
-        /// Save data changes.
+        /// Async save data changes.
         /// </summary>
-        void Save();
+        Task SaveAsync();
```
Comment (MariyaSin) [inline src/Kernel/Database/IBaseDataProvider.cs:17]: need to add and not replace

File: src/Kernel/Validators/ImageContentValidator.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+using System;
+
+namespace LT.DigitalOffice.Kernel.Validators
+{
+    public class ImageContentValidator : AbstractValidator<string>, IImageContentValidator
+    {
+        public ImageContentValidator()
+        {
+            RuleFor(content => content)
+                .NotEmpty().WithMessage("Content can't be empty.");
```
Comment (MariyaSin) [inline src/Kernel/Validators/ImageContentValidator.cs]: i think will be better if you use cascade.Stop and combine conditions

File: src/Kernel/Validators/ImageContentValidator.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+﻿using FluentValidation;
+using LT.DigitalOffice.Kernel.Validators.Interfaces;
+using System;
+
+namespace LT.DigitalOffice.Kernel.Validators
+{
+    public class ImageContentValidator : AbstractValidator<string>, IImageContentValidator
+    {
+        public ImageContentValidator()
```
Comment (MariyaSin) [inline src/Kernel/Validators/ImageContentValidator.cs:9]: may be better validate content and extension here?

File: src/Kernel/Constants/ImageFormats.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections.Immutable;
+
+namespace LT.DigitalOffice.Kernel.Constants
+{
+    public static class ImageFormats
+    {
+        public const string jpgFormat = ".jpg";
+        public const string jpegFormat = ".jpeg";
+        public const string pngFormat = ".png";
```
Comment (Glunky) [inline src/Kernel/Constants/ImageFormats.cs]: without format - just png or jpg

**Review by Glunky — APPROVED** [review]

File: src/Kernel/Database/IBaseDataProvider.cs (hunk 2 above)
Comment (BelehovEgor) [reply src/Kernel/Database/IBaseDataProvider.cs:17]: when do we need void Save()?

**Review by BelehovEgor — APPROVED** [review]

File: src/Kernel/Validators/ImageContentValidator.cs (hunk 4 above)
Comment (ndoriomedov) [reply src/Kernel/Validators/ImageContentValidator.cs:9]: will add another validator for extension in order not to create special model for them both

**Pull request merged**